Marlin/ubl_G29.cpp

```diff
@@ -1438,4 +1461,178 @@
+        SERIAL_PROTOCOLLNPAIR("  grid: ", grid_size_G);
+        return;
+      }
+      
+      // populate X matrix
+      for (G_X_index = 0; G_X_index < grid_size_G; G_X_index++) {
+        grid_G_index_to_xpos[G_X_index] = x_min + G_X_index * (x_max - x_min)/(grid_size_G - 1);
+        if (G_X_index > 0 && grid_G_index_to_xpos[G_X_index - 1] == grid_G_index_to_xpos[G_X_index] ) {
+          grid_G_index_to_xpos[G_X_index] = grid_G_index_to_xpos[G_X_index - 1] + 1;
+        }
+      }
+      
+      // populate Y matrix
+      for (G_Y_index = 0; G_Y_index < grid_size_G; G_Y_index++) {
+        grid_G_index_to_ypos[G_Y_index] = y_min + G_Y_index * (y_max - y_min)/(grid_size_G - 1);
+        if (G_Y_index > 0 && grid_G_index_to_ypos[G_Y_index -1] == grid_G_index_to_ypos[G_Y_index] ) {
+          grid_G_index_to_ypos[G_Y_index] = grid_G_index_to_ypos[G_Y_index - 1] + 1;
+        }
+      }
+      
+      ubl.has_control_of_lcd_panel = true;
+      save_ubl_active_state_and_disable();   // we don't do bed level correction because we want the raw data when we probe
+      
+      DEPLOY_PROBE();
+      
+      // this is a copy of the G29 AUTO_BED_LEVELING_BILINEAR method/code
+      #undef PROBE_Y_FIRST
+      #if ENABLED(PROBE_Y_FIRST)
+        #define PR_OUTER_VAR xCount
+        #define PR_OUTER_NUM grid_size_G
+        #define PR_INNER_VAR yCount
+        #define PR_INNER_NUM grid_size_G
+        #else
+        #define PR_OUTER_VAR yCount
+        #define PR_OUTER_NUM grid_size_G
+        #define PR_INNER_VAR xCount
+        #define PR_INNER_NUM grid_size_G
+      #endif
+      
+      bool zig = PR_OUTER_NUM & 1;  // Always end at RIGHT and BACK_PROBE_BED_POSITION
+      
+      // Outer loop is Y with PROBE_Y_FIRST disabled
+      for (PR_OUTER_VAR = 0; PR_OUTER_VAR < PR_OUTER_NUM; PR_OUTER_VAR++) {
+        
+        int8_t inStart, inStop, inInc;
+        
+SERIAL_PROTOCOLPAIR("\nPR_OUTER_VAR: ", PR_OUTER_VAR);
+
+        if (zig) { // away from origin
+          inStart = 0;
+          inStop = PR_INNER_NUM;
+          inInc = 1;
+        }
+        else {     // towards origin
+          inStart = PR_INNER_NUM - 1;
+          inStop = -1;
+          inInc = -1;
+        }
+        
+        zig = !zig; // zag
+        
+        // Inner loop is Y with PROBE_Y_FIRST enabled
+        for (PR_INNER_VAR = inStart; PR_INNER_VAR != inStop; PR_INNER_VAR += inInc) {
+SERIAL_PROTOCOLPAIR("\nPR_INNER_VAR: ", PR_INNER_VAR);
+
+SERIAL_PROTOCOLPAIR("\nCheckpoint: ", 1);
+          
+          // end of G29 AUTO_BED_LEVELING_BILINEAR method/code
+          if (ubl_lcd_clicked()) {
+SERIAL_PROTOCOLPAIR("\nCheckpoint: ", 2);
+            SERIAL_PROTOCOLLNPGM("\nGrid only partially populated.\n");
+            lcd_quick_feedback();
+            STOW_PROBE();
+SERIAL_PROTOCOLPAIR("\nCheckpoint: ", 3);
+            while (ubl_lcd_clicked()) idle();
+SERIAL_PROTOCOLPAIR("\nCheckpoint: ", 4);
```

Comment:
NO!   I'll get that out of there tomorrow.    PLEASE DON'T EDIT CODE IN THIS AREA.   It will cause conflicts I can't resolve.    And the only reason that is there is because of that Sync crash.   Not everything got perfectly restored.